react/features/base/lib-jitsi-meet/functions.any.js

```diff
@@ -45,6 +45,28 @@ export function isAnalyticsEnabled(stateful: Function | Object) {
     return !(disableThirdPartyRequests || analytics.disabled);
 }
 
+/**
+ * Determines whether an avatar URL is valid.
+ *
+ * @param {Function|Object} stateful - The redux store, state, or
+ * {@code getState} function.
+ * @param {string} avatarURL - The avatar url to check.
+ * @returns {boolean} If the URL matches the origin and disableThirdPartyRequests
+ * is defined, or the URL matches the gravatarBaseURL, {@code true};
+ * {@code false}, otherwise.
+ */
+export function isAvatarUrlValid(stateful: Function | Object, avatarURL: string) {
+    const { disableThirdPartyRequests, gravatarBaseURL } = toState(stateful)['features/base/config'];
+
+    if (disableThirdPartyRequests) {
+        return avatarURL && avatarURL.startsWith(window.location.origin);
+    } else if (gravatarBaseURL) {
+        return avatarURL && avatarURL.startsWith(gravatarBaseURL);
```

Comment:
A participant may have a custom URL even when gravatar is used, this would break that.

Reply:
When a participant sets a custom URL for its avatar, this URL is then fetched by all other participants. This is a security issue since one can make all participants of a jitsi call fetch arbitrary GET requests.
The check on the `gravatarBaseURL` even if `disableThirdPartyRequests` is false would mitigate this issue.

Reply:
That's what `disableThirdPartyRequests` is for. Sorry can't accept this unless you restore the existing behavior.

Reply:
I have restored the existing behavior.